### PR TITLE
*: cherry pick some keyspace related things

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #4399

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by lhy1024** [comment]
ci failed. And do we need to pick "add old keyspace apis for compatibility"?

File: server/apiv2/handlers/keyspace.go [hunk 1]
```diff
@@ -42,6 +42,12 @@ func RegisterKeyspace(r *gin.RouterGroup) {
 	router.PATCH("/:name/config", UpdateKeyspaceConfig)
 	router.PUT("/:name/state", UpdateKeyspaceState)
 	router.GET("/id/:id", LoadKeyspaceByID)
+
+	// The following apis uses custom action and are considered deprecated.
+	router.POST("/:name/update-config", UpdateKeyspaceConfig)
+	router.POST("/:name/enable", EnableKeyspace)
+	router.POST("/:name/disable", DisableKeyspace)
+	router.POST("/:name/archive", ArchiveKeyspace)
```
Comment (AmoebaProtozoa) [inline server/apiv2/handlers/keyspace.go:50]: We probably don't need these

Comment (rleungx) [reply server/apiv2/handlers/keyspace.go:50]: If we are going to use master for serverless, can we remove it?

**Review by nolouch — APPROVED** [review]

**Pull request merged**